### PR TITLE
python3 compatibility

### PR DESCRIPTION
**Pull request by slashmili**
Python doctest was horrible, it was hard to only run specific test and managing. Therefore I moved erl_term tests to nosetests format

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
+python:
+  - 2.7
+  - 3.4
+  - 3.6
+install:
+  - pip install nose
 script:
   - python ./elixir_sense.py
-  - python ./erl_terms.py
+  - nosetests t/erl_terms_test.py
```
Comment (bees) [inline .travis.yml:11]: for what its worth, I believe nosetests is unmaintained and its developers are solely focused on nose2 now. However, we have legacy tests using nosetests where I work and we haven't run into issues yet.

**Comment by bees** [comment]
Looks good to me! :shipit:

File: .travis.yml (hunk 1 above)
Comment (slashmili) [reply .travis.yml:11]: @bees so I should move to nose2! 
Thanks I did feel that nose website is abandoned 😆

Comment (slashmili) [reply .travis.yml:11]: @bees can you take a look at the this one

When I move test in unittest class I get this error.

```
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/milad/.dotfiles/.vim/bundle/alchemist.vim/tests/erl_terms_test.py", line 12, in test_basic_atom
    hello = erl_terms.__encode_atom('hello')
AttributeError: module 'erl_terms' has no attribute '_BasicTermTests__encode_atom'
```

Comment (bees) [reply .travis.yml:11]: will look into it this weekend, sorry about the delay!

Comment (bees) [reply .travis.yml:11]: Finally got to spend time on this and learned something new about python: https://docs.python.org/2.7/tutorial/classes.html#private-variables-and-class-local-references

I'm working on a PR that replaces nose with unittest - I will update the method names so they stop getting mangled.

**Pull request merged**